lib/segment/src/types.rs

```diff
@@ -1026,6 +1026,11 @@ pub struct StrictModeConfig {
     #[serde(skip_serializing_if = "Option::is_none")]
     #[validate(nested)]
     pub sparse_config: Option<StrictModeSparseConfig>,
+
+    /// Max number of payload index in a collection
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[validate(range(min = 1))]
```

Comment:
Why is 1 the limit? Wouldn't be 0 also useful to fully prevent creating indices?